### PR TITLE
Yang la101.energy balance china

### PR DESCRIPTION
**Pull request by thuliuyang**
Add  zchunk_LA101.Energy_Balance_china 
Use the Long format instead of the wide format

### CODE REVIEW

File: R/zchunk_LA101.Energy_Balance_china.R [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+      add_units("Unit = Mtce") %>%
+      add_comments("historical years") %>%
+      add_legacy_name("L101.NBS_use_all_Mtce") %>%
+      add_precursors("gcam-china/province_names_mappings",
+                     "gcam-china/NBS_CESY_process",
+                     "gcam-china/NBS_CESY_material",
+                     "gcam-china/en_balance_Mtce_Yh_province",
+                     "gcam-china/Tibet_share",
+                     "gcam-china/tibet_shares_mappings") %>%
+      add_flags(FLAG_PROTECT_FLOAT, FLAG_SUM_TEST) ->
```
Comment (thuliuyang) [inline R/zchunk_LA101.Energy_Balance_china.R]: I compared the old and new results and found some very small differences after the interpolation. （1e-14）

File: R/zchunk_LA101.Energy_Balance_china.R [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    # We have estimates of Tibet shares of energy use by sector.  A rough estimate to give scale
+    # Tibet accounts for 1% of the national energy.
+    Tibet_share %>%
+      left_join(tibet_shares_mappings, by = c("xz.sector", "xz.fuel")) %>%
+      filter(!grepl('biomass', fuel)) ->
+      L101.tibet_pct
+
+    L101.inNBS_Mtce_province_S_F %>%
+      group_by(year) %>%
+      filter(!is.na(value)) %>%
```
Comment (rynacui) [inline R/zchunk_LA101.Energy_Balance_china.R]: The two lines can be combined as `summarise(value = sum(value), na.rm = T)`

File: R/zchunk_LA101.Energy_Balance_china.R [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+      summarise(value=sum(value)) %>%
+      ungroup() ->
+      L101.inNBS_Mtce_CHINA
+
+    L101.tibet_pct %>%
```
Comment (rynacui) [inline R/zchunk_LA101.Energy_Balance_china.R]: This section seems a little redundant. I'll work with you offline.

File: R/zchunk_LA101.Energy_Balance_china.R [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+      group_by(province,year,sector, fuel) %>%
+      filter(!is.na(value)) %>%
+      summarise(value = sum(value)) %>%
+      ungroup() ->
+      L101.inNBS_Mtce_tibet_S_F
+
+    L101.inNBS_Mtce_province_S_F %>%
+      filter(province != "XZ") %>%
+      bind_rows(L101.inNBS_Mtce_tibet_S_F) %>%
+      complete(nesting(province, sector, fuel), year = c(HISTORICAL_YEARS))  ->
```
Comment (rynacui) [inline R/zchunk_LA101.Energy_Balance_china.R]: Do we need so many `complete`?

File: R/zchunk_LA101.Energy_Balance_china.R [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+      L101.inNBS_Mtce_province_S_F
+
+    # Produce outputs
+    L101.NBS_use_all_Mtce %>%
+      add_title("NBS china energy statistical yearbook by sector / fuel / year") %>%
+      add_units("Unit = Mtce") %>%
+      add_comments("historical years") %>%
```
Comment (rynacui) [inline R/zchunk_LA101.Energy_Balance_china.R]: comments are usually a quick summary of the major steps are taken for each output

File: inst/extdata/gcam-china/province_names_mappings.csv [hunk 6]
```diff
@@ -1,7 +1,7 @@
-# File: province_names_mappings.csv
-# Title: Province alternate names mapping file
-# Units: NA
-# Description: Map province alternate names, administrative region, climate region, and grid region
+# File: province_names_mappings.csv,
```
Comment (rynacui) [inline inst/extdata/gcam-china/province_names_mappings.csv]: Remove the ending `,`

File: R/zchunk_LA101.Energy_Balance_china.R [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+    # Calculate national total first
+    L101.inNBS_Mtce_province_S_F %>%
+      group_by(year) %>%
+      summarise(value = sum(value, na.rm = T)) %>%
+      ungroup() ->
+      L101.inNBS_Mtce_CHINA
+
+    Tibet_share %>%
+      left_join(tibet_shares_mappings, by = c("xz.sector", "xz.fuel")) %>%
+      # TODO: we are removing biomass since all the other provinces do not have it
```
Comment (sypnl) [inline R/zchunk_LA101.Energy_Balance_china.R]: Remove "TODO" . I don't think we need to create a new issue on this.

File: R/zchunk_LA101.Energy_Balance_china.R (hunk 1 above)
Comment (sypnl) [reply R/zchunk_LA101.Energy_Balance_china.R]: @pralitp or @kvcalvin  would this be a problem?

**Comment by rynacui** [comment]
Hi @pralitp, please review this PR when you get a chance. All the other gcam-china level1 chunks depend on this one, so @sypnl and I think it's a good idea to have your approval on this. We can take care of the rest PRs once this one is merged. Those are pretty straightforward. Thank you!!!

**Comment by rynacui** [comment]
Thank you @pralitp!

**Pull request merged**